python/mxnet/io.py

```diff
@@ -305,6 +310,14 @@ def provide_data(self):
                        for r, i in zip(self.rename_data, self.iters)], [])
 
     @property
+    def provide_data_type(self):
```

Comment:
@pluskid 
Do we want to do it this way? If we follow this convention then layout should also be handled like this.

Alternatively we can also put type in to provide_data as a (name, shape, type)